SuperTokensIOS/Classes/Utils.swift

```diff
@@ -171,6 +211,159 @@ internal class Utils {
+    }
+    
+    internal static func saveLastAccessTokenUpdate() {
+        let now = "\(Date().timeIntervalSince1970 * 1000)";
+        
+        storeInStorage(name: SuperTokensConstants.LAST_ACCESS_TOKEN_UPDATE, value: now, expiry: Int.max)
+        
+        storeInStorage(name: "sIRTFrontend", value: "", expiry: 0)
+    }
+    
+    internal static func getFromStorage(name: String) -> String? {
+        let itemInStorage = getUserDefaults().string(forKey: "st-storage-item-\(name)")
+        
+        if itemInStorage == nil {
+            return nil
+        }
+        
+        var parts = itemInStorage!.components(separatedBy: "\(name)=")
+        
+        if parts.count >= 2 {
+            let last = parts.popLast()
+            
+            if last != nil {
+                return last!.components(separatedBy: ";").first
+            }
+        }
+        
+        return nil
+    }
+    
+    internal static func getLocalSessionState() -> LocalSessionState {
+        let lastAccessTokenUpdate = getFromStorage(name: SuperTokensConstants.LAST_ACCESS_TOKEN_UPDATE)
+        let frontTokenExists = FrontToken.doesTokenExist()
+        
+        if frontTokenExists && lastAccessTokenUpdate != nil {
+            return LocalSessionState(status: .EXISTS, lastAccessTokenUpdate: lastAccessTokenUpdate)
+        } else {
+            return LocalSessionState(status: .NOT_EXISTS, lastAccessTokenUpdate: nil)
+        }
+    }
+    
+    internal static func setToken(tokenType: TokenType, value: String) {
+        let name = tokenType.getStorageName()
+        
+        // We save the tokens with a 100-year expiration time
+        return storeInStorage(name: name, value: value, expiry: Int(Date().timeIntervalSince1970 * 1000) + 3153600000)
+    }
+    
+    internal static func saveTokenFromHeaders(httpResponse: HTTPURLResponse) {
+        guard let headerFields: [String: String] = httpResponse.allHeaderFields as? [String: String] else {
+            return
+        }
+        
+        if let refreshToken: String = headerFields[SuperTokensConstants.refreshTokenHeaderKey] {
+            Utils.setToken(tokenType: .refresh, value: refreshToken)
+        }
+        
+        if let accessToken: String = headerFields[SuperTokensConstants.accessTokenHeaderKey] {
+            Utils.setToken(tokenType: .access, value: accessToken)
+        }
+        
+        if let frontToken: String = headerFields[SuperTokensConstants.frontTokenHeaderKey] {
+            FrontToken.setItem(frontToken: frontToken)
+        }
+        
+        if let antiCSRF: String = headerFields[SuperTokensConstants.antiCSRFHeaderKey] {
+            let localSessionState = Utils.getLocalSessionState()
+            
+            if localSessionState.status == .EXISTS {
+                AntiCSRF.setToken(antiCSRFToken: antiCSRF, associatedAccessTokenUpdate: localSessionState.lastAccessTokenUpdate!)
+            }
```

Comment:
why is this happening? Ask @porcellus if you are not sure, and comment in the code why this is happening

Reply:
The website SDK needs it for typing, removed it from this SDK since it handles nil checks in this case